### ISSUE TITLE
Webmention endpoint resolution failing when there are multiple `link`s in header

### ISSUE THREAD

**Issue by campegg** [beatonma/django-wm#41]
When sending a webmention to a site that returns multiple `link`s in the header, the resolved endpoint URL gets mashed together with other `link` values. For example, if the site returns a header like this:

```
HTTP/1.1 200 OK
Date: Mon, 03 Oct 2022 19:33:58 GMT
Content-Type: text/html
Content-Length: 16887
Last-Modified: Sun, 02 Oct 2022 20:46:59 GMT
link: <https://websub.io>; rel="websub"
link: <https://websub.io>; rel="websub"
link: <https://webmention.io>; rel="webmention"
Referrer-Policy: no-referrer
X-Content-Type-Options: nosniff
Accept-Ranges: bytes
```

The URL endpoint resolves as:

```
https://websub.io>; rel="websub",<https://websub.io>; rel="websub",<https://webmention.io
```

I _think_ it has something to do with the regex in the [_get_endpoint_in_http_headers](https://github.com/beatonma/django-wm/blob/846b1e3cd23016c743cc0ed740780814fc34ee84/mentions/tasks/outgoing/remote.py#L157) function; if the string "webmention" is found in the headers, it looks like it might be matching everything from the `https` at start of the first URL until it finds the webmention URL?

**Comment by beatonma** [comment]
Thanks for the report - will be fixed in next release.

**Comment by beatonma** [comment]
Fixed in `3.1.0` - I also noticed that the `WebmentionHeadMiddleware` was overwriting any pre-existing links in the headers so that's been fixed now too.

**Comment by campegg** [comment]
Thank you!
